resolwe_bio/kb/listener_plugin.py

```diff
@@ -25,3 +31,75 @@ class ExposeMapping(ExposeObjectPlugin):
     def filter_objects(self, user: UserClass, queryset: QuerySet, data) -> QuerySet:
         """Filter the objects for the given user."""
         return queryset
+
+
+class FeatureCache(CachedObjectPlugin):
+    model = Feature
+    cached_fields = [
+        "source",
+        "feature_id",
+        "species",
+        "type",
+        "sub_type",
+        "name",
+        "full_name",
+        "description",
+        "aliases",
+    ]
+    identifier_fields = ("feature_id", "source", "species")
+    expiration_time = None  # Never expire.
+
+
+class KnowledgeBasePlugin(ListenerPlugin):
+    """Handler methods for KnowledgeBase methods."""
+
+    name = "KnowledgeBase plugin"
+
+    def handle_filter_features(
+        self,
+        data_id: int,
+        message: Message[str, str, list[str]],
+        manager: "Processor",
+    ) -> Response[list[int]]:
+        """Get the feature data based on the given identifiers."""
+        source, species, feature_ids, requested_fields = message.message_data
```

Comment:
What if there is more that one combination of source and species?

Reply:
Then user will have to perform more than one request. We have discussed this with Janez and he agreed on it.